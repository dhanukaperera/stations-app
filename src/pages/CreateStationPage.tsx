import { useState, FormEvent } from "react";
import { Link } from "react-router-dom";
import { routes } from "../constants/routes";

type FormData = {
	name: string,
	tagline: string,
	streamUrl: string,
	logo: File | null | string
}

const CreateStationPage = () => {

	const [previewUrl, setPreviewUrl] = useState<string | undefined>();

	const handleFormSubmit = (event: FormEvent<HTMLFormElement>) => {
		event.preventDefault()
		const formData = new FormData(event.currentTarget);

		const data: FormData = {
			name: formData.get('name')?.toString() ?? '',
			tagline: formData.get('tagline')?.toString() ?? '',
			streamUrl: formData.get('streamUrl')?.toString() ?? '',
			logo: formData.get('logo') instanceof File ? formData.get('logo') : null,
		};
		console.log(data)
	}

	function handleFileInputChange(event: React.ChangeEvent<HTMLInputElement>) {
		const fileList = event.target.files;

		if (fileList && fileList.length > 0) {
			const fileReader = new FileReader();

			fileReader.onload = function () {
				setPreviewUrl(fileReader.result as string);
			};
			fileReader.readAsDataURL(fileList[0]);
		}

	}


	return <>Create Station Page
		<form onSubmit={handleFormSubmit} >
			<label htmlFor="name">Station Name :<input id="name" type="text" name="name" /> </label>
			<label htmlFor="tagline">Tag Line :	<input id="tagline" type="text" name="tagline" /> </label>
			<label htmlFor="streamUrl" > Steaming URL :	<input id="streamUrl" type="url" name="streamUrl" />
			</label>
			<label htmlFor="logo" >Logo :<input id="logo" name="logo" type="file" accept="image/*" onChange={handleFileInputChange} />
			</label>
			<input type="submit" />
			<button type="reset" onClick={() => setPreviewUrl(undefined)} >Clear</button>
		</form>
		{previewUrl && <img src={previewUrl} width="400px" alt="Preview" />}
		<Link to={routes.HOME} >
			<button>Back</button>
		</Link>
	</>
}

export default CreateStationPage;