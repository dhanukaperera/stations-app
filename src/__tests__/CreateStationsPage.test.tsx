import { expect, test } from 'vitest'
import { render } from '@testing-library/react'
import { StaticRouter } from 'react-router-dom/server';
import CreateStationPage from '../pages/CreateStationPage';


test("it should render correctly the create station page", () => {
	const { asFragment } = render(
		<StaticRouter location={"/"} >
			<CreateStationPage />
		</StaticRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});