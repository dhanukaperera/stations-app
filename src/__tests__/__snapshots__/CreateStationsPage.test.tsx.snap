// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it should render correctly the create station page 1`] = `
<DocumentFragment>
  <div
    class="sc-dmqHEX kcReYj"
  >
    <h1>
      Create Station
    </h1>
    <div
      class="sc-eDDNvR gUUDRM"
    >
      <form
        class="sc-hLseeU gaLath"
      >
        <label
          for="name"
        >
            Name :
          <input
            id="name"
            name="name"
            type="text"
          />
        </label>
        <label
          for="tagline"
        >
          Tag Line :	
          <input
            id="tagline"
            name="tagline"
            type="text"
          />
           
        </label>
        <label
          for="streamUrl"
        >
            URL :	
          <input
            id="streamUrl"
            name="streamUrl"
            type="url"
          />
        </label>
        <label
          for="logo"
        >
          Logo :
          <input
            accept="image/*"
            id="logo"
            name="logo"
            type="file"
          />
        </label>
        <div>
          <input
            type="submit"
          />
        </div>
        <button
          class="sc-jTrPJq hVexkC"
          type="reset"
        >
          Clear
        </button>
      </form>
    </div>
    <a
      href="/"
    >
      <button
        class="sc-jTrPJq hVexkC"
      >
        Back
      </button>
    </a>
  </div>
</DocumentFragment>
`;

exports[`render correctly the create station page 1`] = `
<DocumentFragment>
  <div
    class="sc-dmqHEX kcReYj"
  >
    <h1>
      Create Station
    </h1>
    <div
      class="sc-eDDNvR gUUDRM"
    >
      <form
        class="sc-hLseeU gaLath"
      >
        <label
          for="name"
        >
            Name :
          <input
            id="name"
            name="name"
            type="text"
          />
        </label>
        <label
          for="tagline"
        >
          Tag Line :	
          <input
            id="tagline"
            name="tagline"
            type="text"
          />
           
        </label>
        <label
          for="streamUrl"
        >
            URL :	
          <input
            id="streamUrl"
            name="streamUrl"
            type="url"
          />
        </label>
        <label
          for="logo"
        >
          Logo :
          <input
            accept="image/*"
            id="logo"
            name="logo"
            type="file"
          />
        </label>
        <div>
          <input
            type="submit"
          />
        </div>
        <button
          class="sc-jTrPJq hVexkC"
          type="reset"
        >
          Clear
        </button>
      </form>
    </div>
    <a
      href="/"
    >
      <button
        class="sc-jTrPJq hVexkC"
      >
        Back
      </button>
    </a>
  </div>
</DocumentFragment>
`;

exports[`renders correctly with some stations 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB dzAcBK"
  >
    <img
      alt="Capital Teesside logo"
      src="https://herald.musicradio.com/media/0d3d891d-32f2-4c53-95ee-8d1e35bdd126.png"
    />
    <div
      class="sc-gueYoa kEobtL"
    >
      <h1>
        Capital Teesside
      </h1>
      <p>
        The UK's No.1 Hit Music Station
      </p>
      <audio
        class="react-audio-player "
        controls=""
        id=""
        preload="metadata"
        src="https://media-ssl.musicradio.com/CapitalTeesside"
        title="https://media-ssl.musicradio.com/CapitalTeesside"
      >
        <p>
          Your browser does not support the 
          <code>
            audio
          </code>
           element.
        </p>
      </audio>
    </div>
  </div>
</DocumentFragment>
`;
